lib/pages/dashboard_page.dart

```diff
@@ -206,6 +209,61 @@ class _DashboardPageState extends State<DashboardPage> with WindowListener {
     });
 
     Future(() => _checkForUpdates(notifyIfLatest: false, notifyIfError: false));
+
+    var notifications = ntConnection.subscribe("notifications");
+
+    //dont display a notification when we first connect
+    bool notificationFirstRun = true;
+    notifications.listen((p0, p1) {
+      List<String> data = p0.toString().replaceAll("[", "").replaceAll("]", "").split(",");
+      if(notificationFirstRun) {
+        notificationFirstRun = false;
+        return;
+      }
+      Icon icon;
+      if(data[0] == "INFO") {
+        icon = const Icon(Icons.info);
+      }
+      else if(data[0] == "WARNING") {
+        icon = const Icon(Icons.warning_amber, color: Colors.orange,);
+      }
+      else if(data[0] == "ERROR") {
+        icon = const Icon(Icons.error, color: Colors.red,);
+      }
+      else {
+        icon = const Icon(Icons.question_mark);
+      }
+
+      String title = data[1];
+      String description = data[2];
+      setState(() {
+        ColorScheme colorScheme = Theme.of(context).colorScheme;
+        TextTheme textTheme = Theme.of(context).textTheme;
+        ElegantNotification notification = ElegantNotification(
+          autoDismiss: true,
+          showProgressIndicator: true,
+          background: colorScheme.surface,
+          width: 350,
+          position: Alignment.bottomRight,
+          title: Text(
+            title,
+            style: textTheme.bodyMedium!.copyWith(
+              fontWeight: FontWeight.bold,
+            ),
+          ),
+          icon: icon,
+          description: Text(description),
+          stackedOptions: StackedOptions(
+            key: 'notification',
+            type: StackedType.above,
+            itemOffset: const Offset(0, 5),
+          ),
+        );
+        if (mounted) {
+          notification.show(context);
+        }
+      });
+    });
```

Comment:
I think the listening logic would belong better in a separate class that you pass a function into which is called when a new notification is published. The main reason is because it could eventually cause the method to become a bit cluttered.

Reply:
Would a robot alert service similar be a good way of going about this?